src/SAWScript/Crucible/Common/Setup/Builtins.hs

```diff
@@ -83,3 +84,19 @@ crucible_execute_func args = do
                                                   | a <- args
                                                   | t <- tps
                                                   ]
+
+--------------------------------------------------------------------------------
+-- ** Shared data types
+
+-- | When invoking a points-to command, against what should SAW check the type
+-- of the RHS value?
+data CheckPointsToType ty
+  = CheckAgainstPointerType
+    -- ^ Check the type of the RHS value against the type that the LHS points to.
+    --   Used by commands such as @llvm_{conditional_}points_to@ and
+    --   @mir_points_to@.
+  | CheckAgainstCastedType ty
+    -- ^ Check the type of the RHS value against the provided @ty@, which
+    --   the LHS pointer is casted to.
+    --   This is currently used by @llvm_{conditional_}points_to_at_type@ only.
+  deriving (Functor, Foldable, Traversable)
```

Comment:
Reviewers' note: this isn't new code, but rather code that was moved from `SAWScript.Crucible.LLVM.Builtins`. This data type is now used in the MIR-related code in `saw-remote-api`, so it no longer made sense for it to live in an LLVM-specific module.